homeassistant/core.py

```diff
@@ -918,6 +918,24 @@ def async_entity_ids(
             if state.domain in domain_filter
         ]
 
+    @callback
+    def async_entity_ids_count(
+        self, domain_filter: Optional[Union[str, Iterable]] = None
+    ) -> int:
+        """Count the entity ids that are being tracked.
+
+        This method must be run in the event loop.
+        """
+        if domain_filter is None:
+            return len(self._states.keys())
```

Comment:
Taking the length of a dictionary counts the keys (items) by default.

Reply:
Fixed in the next PR https://github.com/home-assistant/core/pull/40272